main.js

```diff
@@ -99,6 +99,40 @@ function displayPalette() {
   }
 }
 
+function dragstartHandler(event) {
+  event.dataTransfer.setData('color-box', event.target.getAttribute("data-id"));
+  event.dataTransfer.effectAllowed = "move";
+}
+
+function dragoverHandler(event) {
+  event.preventDefault();
+  event.dataTransfer.dropEffect = "move";
+}
+
+function dropHandler(event) {
+  event.preventDefault();
+  var dragId = parseInt(event.dataTransfer.getData('color-box'));
+  var dropId = parseInt(event.target.parentNode.dataset.id);
+  var dragColor;
+  var dropColor;
+  var dragIndex;
+  var dropIndex;
+    for (var i = 0; i < currentPalette.colors.length; i++) {
+      if(currentPalette.colors[i].id === dragId){
+        dragColor = currentPalette.colors[i];
+        dragIndex = i;
+      }
+      if(currentPalette.colors[i].id === dropId){
+        dropColor = currentPalette.colors[i];
+        dropIndex = i;
+      }
+    }
+    currentPalette.colors[dropIndex] = dragColor;
+    currentPalette.colors[dragIndex] = dropColor;
+
+    displayPalette();
+}
```

Comment:
If anyone thinks up a way to refactor this to be more concise (especially declaring all those variables at the top) I'd love to hear it